### ISSUE TITLE
Data definition for Line charts should be a plain array of lineChartDataByTopic

### ISSUE THREAD

**Issue by AlfonsoML** [eventbrite/britecharts#250]
The [LineChartData ](http://eventbrite.github.io/britecharts/global.html#LineChartData__anchor)is defined as an object with a single property dataByTopic that it's the array of lineChartDataByTopic

That's not really needed. The dataset should be just the array of topics, avoiding to have an extra element around.

## Expected Behavior
Just use an array of topics as the dataset for line charts.

## Current Behavior
The dataset is an object with a single property that it's the real array of data.

## Possible Solution
This change would break back compatibility, but it's easy to adapt. Those that didn't have their data with the dataByTopic property can get rid of it. Those that have that kind of data, then use dataset.dataByTopic object as the input for the chart.

change from
```
{
    dataByTopic: [
        {
            topicName: 'San Francisco',
            topic: 123,
            dates: [
                {
                    date: '2017-01-16T16:00:00-08:00',
                    value: 1
                },
                {
                    date: '2017-01-16T17:00:00-08:00',
                    value: 2
                }
            ]
        },
        {
            topicName: 'Other',
            topic: 345,
            dates: [
                {...},
                {...}
            ]
        }
    ]
}
```
to
```

[
        {
            topicName: 'San Francisco',
            topic: 123,
            dates: [
                {
                    date: '2017-01-16T16:00:00-08:00',
                    value: 1
                },
                {
                    date: '2017-01-16T17:00:00-08:00',
                    value: 2
                }
            ]
        },
        {
            topicName: 'Other',
            topic: 345,
            dates: [
                {...},
                {...}
            ]
        }
    ]
```

**Comment by Golodhros** [comment]
Thanks @AlfonsoML,

That's legit! At first we were passing two objects, one by date and another by topic, but now we are doing the proper calculations inside our charts, so we don't need them.

This structure is a legacy of those times, so YES, we should simplify it as you exactly pointed out!

Thanks for the issue!

**Comment by Golodhros** [comment]
That would make sense @DamonCool, this flat data schema is being used in the stacked area chart as well. We should allow it.

**Comment by Golodhros** [comment]
Fixed in #713
